source/traction_boundary_conditions/lithostatic_pressure.cc

```diff
@@ -0,0 +1,469 @@
+            AssertThrow(representative_point[d] <= extents[d], ExcMessage("Coordinate "
+                                                                          + dealii::Utilities::int_to_string(d)
+                                                                          +"Representative point outside box domain."));
+
+          // set z of surface
+          representative_point[dim-1] = extents[dim-1];
+        }
+      else
+        AssertThrow(false, ExcNotImplemented());
+
+      // set up the input for the density function of the material model
+      typename MaterialModel::Interface<dim>::MaterialModelInputs in0(1, n_compositional_fields);
+      typename MaterialModel::Interface<dim>::MaterialModelOutputs out0(1, n_compositional_fields);
+
+      // where to calculate the density
+      // for spherical domains
+      if (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) != 0 ||
+          dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()) != 0 ||
+          dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()) != 0 ||
+          /*dynamic_cast<const GeometryModel::TwoMergedChunks<dim>*> (&this->get_geometry_model()) != 0*/
+          dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*> (&this->get_geometry_model()) != 0 )
+        {
+          // set spherical surface coordinates
+          in0.position[0] = spherical_to_cart(spherical_representative_point);
+        }
+      // and for cartesian domains
+      else
+        {
+          // set cartesian surface coordinates
+          in0.position[0] = representative_point;
+        }
+
+      // we need the actual temperature at t0
+      in0.temperature[0] = this->get_initial_conditions().initial_temperature(in0.position[0]);
+
+      // the previous pressure
+      in0.pressure[0] = pressure[0];
+
+      // we need the actual composition at t0
+      for (unsigned int c=0; c<n_compositional_fields; ++c)
+        in0.composition[0][c] = this->get_compositional_initial_conditions().initial_composition(in0.position[0], c);
+
+      // we do not need the viscosity
+      in0.strain_rate.resize(0);
+
+      // evaluate
+      this->get_material_model().evaluate(in0, out0);
+
+      // get the magnitude of gravity. we assume
+      // that gravity always points along the depth direction. this
+      // may not strictly be true always but is likely a good enough
+      // approximation here.
+      const double density0 = out0.densities[0];
+      const double gravity0 = this->get_gravity_model().gravity_vector(in0.position[0]).norm();
+
+      // now integrate pressure downward using trapezoidal integration
+      // p'(z) = rho(p,c,T) * |g| * delta_z
+      double sum = delta_z * 0.5 * density0 * gravity0;
+
+      // we assume that the t0 temperature and composition fields at the boundary
+      // are representative throughout the model time and model domain
```

Comment:
I assume using the pressure from the first time step works for the models you want to run? You could also use the `update()` function, and then integrate using the temperature/composition from the current time step. That would be more expensive, but you could make it optional. 
Or would that be something that is part of the other plugin you wanted to add?